CODING_GUIDLINES.md

```diff
@@ -0,0 +1,119 @@
+- Transpiler: [Babel](https://babeljs.io/) + [TypeScript](https://www.typescriptlang.org/)
+
+## Requirements
+
+Cosmos Explorer has been under constant development for over 5 years. As a result there are many different patterns and practices in the code base. This document serves as a guide to how we write code and helps avoid propagating practices which are no longer preferred. Each requirement in this document is labelled and color-coded to show the relative importance. In order from highest importance to lowest importance:
+
+✅ DO this. If you feel you need an exception, engage with the project owners _prior_ to implementation.
+
+⛔️ DO NOT do this. If you feel you need an exception, engage with the project owners _prior_ to implementation.
+
+☑️ YOU SHOULD strongly consider this but it is not a requirement. If not following this advice, please comment code with why and proactively begin a discussion as part of the PR process.
+
+⚠️ YOU SHOULD NOT strongly consider not doing this. If not following this advice, please comment code with why and proactively begin a discussion as part of the PR process.
+
+💭 YOU MAY consider this advice if appropriate to your situation. Other team members may comment on this as part of PR review, but there is no need to be proactive.
+
+## Development Environment
+
+☑️ YOU SHOULD use VSCode and install the following extensions:
+
+- [Prettier](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
+
+This setup will catch most linting/formatting/type errors as you develop.
+
+☑️ YOU SHOULD use the [GitHub CLI](https://cli.github.com/). It has helpful workflows for submitting PRs as well as for checking out other team members PRs.
+
+💭 YOU MAY use Windows, Linux (including WSL), or OSX. We have team members on all three environments.
+
+✅ DO maintain cross platform compatability when modifying any engineering systems.
+
+## Formatting
+
+✅ DO use [Prettier](https://prettier.io/) to format your code. This will be preformed automatically if using the recommended editor setup. Otherwise `npm run format` will also format code.
+
+## Linting
+
+✅ DO use [ESLint](https://eslint.org/) to check for linting errors your code. This can also be performed by `npm run lint`
+
+💭 YOU MAY consider adding new lint rules. If you find yourself performing "nits" as part of PR review, consider if it is possible to add a lint rule that will autuomatically catch the error in the future
+
+⚠️ YOU SHOULD NOT disable any lint rules. They are there as guidance and to catch common mistakes. You will find places we disable specific lint rules, but is should be exceptional. If a rule does need to be disabled, prefer disabling a specific line instead of the entire file.
+
+⛔️ DO NOT add any tslint rules. TSLint has been deprecated and is on track to be removed. Always prefer ESLint rules.
+
```

Comment:
Can we group them by category? All do this together, all do nots together , etc for better readability

Reply:
How about grouping the categories inside each component? IE all the `DOs` go together under linting, then under UI, etc. I think keeping a global `DOs` list will be worse for people finding stuff